docs/book/src/images/control-plane-machines-resources.plantuml

```diff
@@ -0,0 +1,44 @@
+@startuml control-plane-machines-resources
+
+skinparam object {
+    backgroundColor<< Infrastructure provider CRD >> #D9EAD3
+}
+
+' Cluster API CRD
+object Cluster << Cluster API CRD >> {
+    <font color="Green">spec.infrastructureRef</font>
+    spec.controlPlaneRef
+}
+
+object MachineHealthCheck << Cluster API CRD >> {
+    spec.clusterName
+}
+
+object KubeadmControlPlane << Cluster API CRD >> {
+    <font color="Green">spec.infrastructureTemplate</font>
+    spec.kubeadmConfigSpec
+}
+
+object Machine << Cluster API CRD >> {
+    <font color="Green">spec.infrastructureRef</font>
+    spec.bootstrap.configRef
+}
+
+object KubeadmConfig << Cluster API CRD >> 
+
+' Infrastructure provider CRD
+object FooCluster << Infrastructure provider CRD >>
+object FooMachineTemplate << Infrastructure provider CRD >>
+object FooMachine << Infrastructure provider CRD >>
+
+MachineHealthCheck "0..*" .right.>  "1" Cluster
```

Comment:
The spacing between the 0..* -> 1 relationship in the png doesn't maintain the appropriate context.

There should also likely be a relationship shown between MachineHelathCheck and the Machines managed by KubeadmControlPlane, but any relationship between MachineHealthCheck and KubeadmControlPlane should likely not be merged until https://github.com/kubernetes-sigs/cluster-api/pull/3185 has merged to add support.

Reply:
I added a comment between MachineHealthCheck and KubeadmControlPlane at 49bd94548.